Libraries/CustomComponents/NavigationExperimental/NavigationHeader.js

```diff
@@ -27,123 +27,227 @@
  */
 'use strict';
 
-const Animated = require('Animated');
-const Image = require('Image');
-const NavigationContainer = require('NavigationContainer');
-const NavigationRootContainer = require('NavigationRootContainer');
 const React = require('react-native');
-const StyleSheet = require('StyleSheet');
-const Text = require('Text');
-const TouchableOpacity = require('TouchableOpacity');
-const View = require('View');
+const NavigationContainer = require('NavigationContainer');
+const NavigationHeaderTitle = require('NavigationHeaderTitle');
+const NavigationHeaderBackButton = require('NavigationHeaderBackButton');
+
+const {
+  Animated,
+  Platform,
+  StyleSheet,
+  View,
+} = React;
+
+const APPBAR_HEIGHT = Platform.OS === 'ios' ? 44 : 56;
+const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : 0;
 
 import type {
-  NavigationState,
-  NavigationParentState
+  NavigationScene,
 } from 'NavigationStateUtils';
 
+type Renderer = (scene: NavigationScene) => ReactElement;
+
+type DefaultProps = {
+  renderTitleComponent: Renderer;
+  renderLeftComponent: Renderer;
+};
+
 type Props = {
-  navigationState: NavigationParentState,
-  onNavigate: Function,
   position: Animated.Value,
-  getTitle: (navState: NavigationState) => string,
+  scenes: Array<NavigationScene>;
+  index: number;
+  renderTitleComponent: Renderer;
+  renderLeftComponent: Renderer;
+  renderRightComponent: Renderer;
+  style: any,
 };
 
-class NavigationHeader extends React.Component {
-  _handleBackPress: Function;
-  props: Props;
-  componentWillMount() {
-    this._handleBackPress = this._handleBackPress.bind(this);
+class NavigationHeader extends React.Component<DefaultProps, Props, void> {
```

Comment:
Could you have a wrong version of flow or react installed maybe?
This would keep flow happy on my PC
```
class NavigationHeader extends React.Component {
  static APPBAR_HEIGHT = APPBAR_HEIGHT;
  static STATUSBAR_HEIGHT = STATUSBAR_HEIGHT;
  static defaultProps: Object;
```
But obviously this is not your intention.